consent/api/v1/views.py

```diff
@@ -103,13 +103,31 @@ def get_required_query_params(self, request):
 
         """
         if request.method in self.QUERY_PARAM_METHODS:
-            username = request.query_params.get(self.REQUIRED_PARAM_USERNAME, '')
-            course_id = request.query_params.get(self.REQUIRED_PARAM_COURSE_ID, '')
-            enterprise_customer_uuid = request.query_params.get(self.REQUIRED_PARAM_ENTERPRISE_CUSTOMER)
+            username = request.query_params.get(
+                self.REQUIRED_PARAM_USERNAME,
+                request.data.get(self.REQUIRED_PARAM_USERNAME, '')
+            )
+            course_id = request.query_params.get(
+                self.REQUIRED_PARAM_COURSE_ID,
+                request.data.get(self.REQUIRED_PARAM_COURSE_ID, '')
+            )
+            enterprise_customer_uuid = request.query_params.get(
+                self.REQUIRED_PARAM_ENTERPRISE_CUSTOMER,
+                request.data.get(self.REQUIRED_PARAM_ENTERPRISE_CUSTOMER)
+            )
         else:
-            username = request.data.get(self.REQUIRED_PARAM_USERNAME, '')
-            course_id = request.data.get(self.REQUIRED_PARAM_COURSE_ID, '')
-            enterprise_customer_uuid = request.data.get(self.REQUIRED_PARAM_ENTERPRISE_CUSTOMER)
+            username = request.data.get(
+                self.REQUIRED_PARAM_USERNAME,
+                request.query_params.get(self.REQUIRED_PARAM_USERNAME, '')
+            )
+            course_id = request.data.get(
+                self.REQUIRED_PARAM_COURSE_ID,
+                request.query_params.get(self.REQUIRED_PARAM_COURSE_ID, '')
+            )
+            enterprise_customer_uuid = request.data.get(
+                self.REQUIRED_PARAM_ENTERPRISE_CUSTOMER,
+                request.query_params.get(self.REQUIRED_PARAM_ENTERPRISE_CUSTOMER)
+            )
```

Comment:
Different HTTP methods have different preferred methods for transmitting information - either as query params or in the body. POST and GET both have well-defined canonical methods, but DELETE has different behaviors assigned to it by different people. Slumber, in particular, expects to send parameters on DELETE via query param, but we were previously expecting it in the body, so that was causing issues with the clients in platform and ecom.

So, what we now do here is look at where we think details should be _first_, but _also_ check the other method to make sure we don't miss anything.